xray/core/dataset.py

```diff
@@ -804,17 +804,26 @@ def reset_coords(self, names=None, drop=False, inplace=False):
                 del obj._variables[name]
         return obj
 
-    def dump_to_store(self, store, encoder=None, sync=True):
+    def dump_to_store(self, store, encoder=None, sync=True, encoding={}):
         """Store dataset contents to a backends.*DataStore object."""
         variables, attrs = conventions.encode_dataset_coordinates(self)
+
+        check_encoding = set()
+        for k, enc in encoding.items():
```

Comment:
This is a more general question about how we want to use `items()` and `iteritems()`.  We define a function `iteritems` in `compat.py` so we get the same behavior (iterator) in Python 2 and 3. Is there a reason we are only using it sometimes but not others?